src/integrationTest/java/integration/tests/SystemEngineTest.java

```diff
@@ -189,6 +198,53 @@ void useFailure() throws SQLException {
 		}
 	}
 
+	@Test
+	@Tag("v2")
+	void connectToAccountWithoutUser() throws SQLException, IOException {
+		ConnectionInfo current = integration.ConnectionInfo.getInstance();
+		String database = current.getDatabase();
+		String serviceAccountName = format("%s_sa_no_user_%d", database, System.currentTimeMillis());
+		try (Connection connection = createConnection(getSystemEngineName())) {
+			try {
+				connection.createStatement().executeUpdate(format("CREATE SERVICE ACCOUNT \"%s\" WITH DESCRIPTION = \"Ecosytem test with no user\"", serviceAccountName));
+				// This what I want to do here
+//				ResultSet genKeyRs = connection.createStatement().executeQuery(format("CALL fb_GENERATESERVICEACCOUNTKEY('%s')", serviceAccountName));
+//				assertTrue(genKeyRs.next());
+//				String clientId = genKeyRs.getString(2);
+//				String clientSecret = genKeyRs.getString(3);
+				// But response of this command is incorrect (FIR-28997), so we have to retrieve clientId and clientSecret using SELECT
+				FireboltConnection fbConn = (FireboltConnection)connection;
+				String accessToken = fbConn.getAccessToken().orElseThrow(() -> new IllegalStateException("access token is not found"));
+				FireboltProperties fbProps = fbConn.getSessionProperties();
+				URL url = new URL(format("%s/query?output_format=TabSeparatedWithNamesAndTypes&database=%s&account_id=%s", fbProps.getHttpConnectionUrl(), current.getDatabase(), fbProps.getAccountId()));
+				HttpURLConnection con = (HttpURLConnection)url.openConnection();
+				con.setRequestMethod("POST");
+				con.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+				con.setRequestProperty("authorization", "Bearer " + accessToken);
+				con.setDoOutput(true);
+				try (PrintStream ps = new PrintStream(con.getOutputStream())) {
+					ps.println(format("CALL fb_GENERATESERVICEACCOUNTKEY('%s')", serviceAccountName));
+				}
+				BufferedReader br = new BufferedReader(new InputStreamReader(con.getInputStream(), UTF_8));
+				String[] data = br.readLine().split("\\t");
+//				String clientId = data[1];
+				String clientSecret = data[2];
+				// end of patch against FIR-28997
```

Comment:
Maybe let's put this hack in a separate function to keep the test more or less readable.